File: soyeon/[3] Programmers/타겟 넘버.java
class Solution {
	public static int answer=0;
    public int solution(int[] numbers, int target) {
    	findAll(numbers, target, 1, numbers[0]);
    	findAll(numbers, target, 1, numbers[0]*(-1));
        return answer;
    }
    public void findAll(int[] numbers, int target, int nowIndex, int sum) {
    	if(nowIndex==numbers.length) {
    		if(target==sum) {
    			answer++;
    		}
    		return;
    	}
    	findAll(numbers, target, nowIndex+1, sum+numbers[nowIndex]);
    	findAll(numbers, target, nowIndex+1, sum-numbers[nowIndex]);
    }
}

File: kuyhochung/1. 매주 문제풀이/41주차/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void dfs(vector<int> numbers, int target, int sum, int count) {
    if (count == numbers.size()) {
        if (sum == target) answer++;
        return;
    }
    
    dfs(numbers, target, sum + numbers[count], count+1);
    dfs(numbers, target, sum - numbers[count], count+1);
}

int solution(vector<int> numbers, int target) {
    dfs(numbers, target, 0, 0);
    return answer;
}